GitCommands/Settings/AppSettings.cs

```diff
@@ -2128,6 +2127,22 @@ private static void SaveEncodings()
             SetString("AvailableEncodings", availableEncodings);
         }
 
+        private static bool? _isDesignMode;
+
+        private static bool IsDesignMode
+        {
+            get
+            {
+                if (_isDesignMode is null)
+                {
+                    string processName = Process.GetCurrentProcess().ProcessName.ToLowerInvariant();
+                    _isDesignMode = processName.Contains("devenv") || processName.Contains("designtoolsserver");
+                }
+
+                return _isDesignMode.Value;
+            }
+        }
```

Comment:
@KlausLoeffelmann this seems the only viable way to detect the designer I found.